.travis.yml

```diff
@@ -9,10 +16,10 @@ stages:
   - test
   - name: deploy
     if: type != pull_request
+script: make verify build build-integration build-e2e test images svcat
 jobs:
   include:
-    # Test
-    - script: make verify build build-integration build-e2e test images svcat
+    # Test is implicit from the build matrix
```

Comment:
not that familiar with travis config files. when you moved the `script` directive up a level, does it get executed with each env var in the matrix?

Reply:
scripts defined in `jobs` section are over rides for whatever was defined at the top level. As we did not define anything at the top level, it defaulted to a ruby job.

So we ended up with 4 jobs:
1. default ruby with go = 1.9
2. default ruby with go = rc
3. test stage with first matrix inherited go = 1.9
4. deploy with first matrix inherited go = 1.9

Moving it up to the toplevel makes it the default job, at which point it knows what kind of env we have and schedules appropriately. The first job assumes the default script, and I think deploy is overwriting the default script to do it's own thing. so we end up with:
- default is defined, and delegated to test, a matrix is defined, run test stage with default script go=1.9
- default is defined, and delegated to test, a matrix is defined, run test stage with default script go=rc
- deploy stage has own definitions of everything, inherits first of matrix go=1.9